src/backend/x11/mod.rs

```diff
@@ -162,8 +163,30 @@ impl X11Backend {
         let (connection, screen_number) = XConnection::new(&logger)?;
         let connection = Arc::new(connection);
         info!(logger, "Connected to screen {}", screen_number);
-
         let xcb = connection.xcb_connection();
+
+        // Does the X server support dri3?
+        let (dri3_major, dri3_minor) = {
+            let extension = xcb.query_extension(dri3::X11_EXTENSION_NAME.as_bytes())?.reply()?;
```

Comment:
I'd recommend `xcb.extension_information(dri3::X11_EXTENSION_NAME)?.is_some()`. x11rb needs the QueryExtension reply as well to send the following `DRI3` `QueryVersion` request (the reply contains the major opcode which is a byte that must be set in the request). If you use `extension_information()`, x11rb caches this reply and saves one round-trip later when it does not have to request the same information again.

(And no, this does not really make any difference. At all. One round-trip should be quite fast. So, I guess this is mostly FYI.)